openslides/core/config.py

```diff
@@ -78,50 +80,38 @@ def __setitem__(self, key, value):
                 raise ConfigError(e.messages[0])
 
         # Save the new value to the database.
-        config_store, created = ConfigStore.objects.get_or_create(key=key, defaults={'value': value})
-        if not created:
-            config_store.value = value
-            config_store.save()
-
-        # Update cache.
-        if hasattr(self, '_cache'):
-            self._cache[key] = value
+        ConfigStore.objects.update_or_create(key=key, defaults={'value': value})
 
         # Call on_change callback.
         if config_variable.on_change:
             config_variable.on_change()
 
-    def items(self):
+    def update_config_varialbes(self, items):
         """
-        Returns key-value pairs of all config variables.
+        Updates the config_variables dict.
+
+        items has to be an iterator over ConfigVariable objects.
         """
-        if not hasattr(self, '_cache'):
-            self.setup_cache()
-        return self._cache.items()
+        self.config_variables.update((variable.name, variable) for variable in items)
```

Comment:
Please add a check that the elements are unique. Raise `ConfigError(_('Too many values for config variable %s found.') % config_variable.name)` for this.